wallet/update.go

```diff
@@ -170,26 +170,26 @@ func appliedEvents(cau chain.ApplyUpdate, walletAddress types.Address) (events [
 	}
 
 	// add the file contract outputs
-	cau.ForEachFileContractElement(func(fce types.FileContractElement, created bool, rev *types.FileContractElement, resolved bool, valid bool) {
-		if !resolved {
-			return
+	for _, fced := range cau.FileContractElementDiffs() {
+		if !fced.Resolved {
+			continue
 		}
-
+		fce := fced.FileContractElement
 		fce.StateElement.MerkleProof = nil // clear the proof to save space
 
-		if valid {
+		if fced.Valid {
 			for i, so := range fce.FileContract.ValidProofOutputs {
 				if so.Address != walletAddress {
 					continue
 				}
 
-				outputID := types.FileContractID(fce.ID).ValidOutputID(i)
+				outputID := fce.ID.ValidOutputID(i)
 				sce, ok := siacoinElements[outputID]
 				if !ok {
 					panic("missing siacoin element")
 				}
 
-				addEvent(types.Hash256(types.FileContractID(fce.ID).ValidOutputID(0)), EventTypeV1ContractResolution, EventV1ContractResolution{
+				addEvent(types.Hash256(fce.ID.ValidOutputID(0)), EventTypeV1ContractResolution, EventV1ContractResolution{
```

Comment:
Ditto. This should be `ValidOutputID(i)` or just `outputID`. Wanted to note it, don't mind resolving in a follow up.

Reply:
good catch, fixed in #168